src/query/storages/fuse/src/operations/common/processors/sink_commit.rs

```diff
@@ -441,12 +447,6 @@ where F: SnapshotGenerator + Send + 'static
                         if let Some(files) = &self.copied_files {
                             metrics_inc_commit_copied_files(files.file_info.len() as u64);
                         }
-                        for segment in self.abort_operation.segments.iter() {
-                            self.ctx.add_segment_location((
-                                segment.to_string(),
-                                SegmentInfo::VERSION,
-                            ))?;
```

Comment:
I am not sure if this is safe, seems agg index or something else depends on this?

Reply:
> I am not sure if this is safe, seems agg index or something else depends on this?

This operation is safe, we have added the add_segment_location operation where new segments are generated.
For example: https://github.com/datafuselabs/databend/pull/15557/files#diff-7c640d95f5c24d091ee74645f89c05486bee365406e05fe6168483b3a6432edaR234-R235